pkg/vault/contracts/AddUnbalancedLiquidityViaSwapRouter.sol

```diff
@@ -138,28 +140,25 @@ contract AddUnbalancedLiquidityViaSwapRouter is RouterQueries, IAddUnbalancedLiq
      */
     function _computeAddUnbalancedLiquidityViaSwap(
         AddLiquidityAndSwapHookParams calldata hookParams
-    ) private returns (IERC20[] memory tokens, uint256[] memory amountsIn, uint256[] memory amountsOut) {
-        // maxAmountsIn length is checked against tokens length at the Vault.
+    ) private returns (IERC20[] memory tokens, uint256[] memory amountsIn) {
         tokens = _vault.getPoolTokens(hookParams.pool);
-        amountsOut = new uint256[](tokens.length);
-        uint256[] memory maxAmountsIn = new uint256[](tokens.length);
 
-        uint256 exactTokenIndex = hookParams.operationParams.exactToken == tokens[0] ? 0 : 1;
-        uint256 adjustableTokenIndex = exactTokenIndex == 0 ? 1 : 0;
-
-        maxAmountsIn[exactTokenIndex] = hookParams.operationParams.maxAdjustableAmount;
-        maxAmountsIn[adjustableTokenIndex] = hookParams.operationParams.maxAdjustableAmount;
+        (uint256 exactTokenIndex, uint256 adjustableTokenIndex) = hookParams.operationParams.exactToken == tokens[0]
+            ? (0, 1)
+            : (1, 0);
 
         (amountsIn, , ) = _vault.addLiquidity(
             AddLiquidityParams({
                 pool: hookParams.pool,
                 to: hookParams.sender,
-                maxAmountsIn: maxAmountsIn,
+                maxAmountsIn: [_MAX_AMOUNT, _MAX_AMOUNT].toMemoryArray(),
```

Comment:
hm, this is a test function.
Perhaps it's ok to use in prod, but we should move it out of the `test` directory if so.

Otherwise just create an array and populate it manually.